week4_HW/mytodo-app/src/components/TodoList.js

```diff
@@ -0,0 +1,46 @@
+import React, { Component } from 'react'
+import input from './buttons'
+import InsertTasks from './insertTasks'
+import {inject, observer} from 'mobx-react'
+
+@inject("data")
+@observer
+export default class TodoList extends Component{
+    render() {
+        console.log(this.props.data)
+        const { deletingTask, checkingBox, addingNewTask} = this.props.data
+        const myTasks = this.props.data.tasksData.map((task, index) => {
+            if (task.done === true) {
+                return (
+                    <li key={task.id}
+                        className="unDoneTask"
+                        style={{ "textDecoration": "line-through" }}>
+                        <input type="checkbox"
```

Comment:
This approach is broken. You have an uncontrolled input here with an onChange handler. You are not controlling the checked value of the input directly. The only reason this works is because you're using `defaultChecked`, but this is not a safe approach. If you are setting the "done" status of your todos to `false` automatically, you already have everything you need to create a proper controlled input.

It seems like you are switching over `task.done` and rendering big pieces of JSX based on this checked value. Instead I'd like to see conditional rendering inside your JSX, something like:

    <li key={task.id}
        className={task.done ? "" : "unDoneTask"}
    >
        <input type="checkbox"
            onChange={() => checkingBox(index)}
            checked={task.done} />
    {index+1}- {task.description} on {task.deadline}
        <input type="button" value="Remove"
            onClick={() => deletingTask(index)} />
    </li>